api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java

```diff
@@ -908,12 +911,9 @@ private static String generateNotebookUserCode(
     String namespace = "dataset_" + qualifier + "_" + domainAsString + "_";
     // Comments in R and Python have the same syntax
     String descriptiveComment =
-        new StringBuilder("# This query represents dataset \"")
-            .append(dataSetName)
-            .append("\" for domain \"")
-            .append(domainAsString)
-            .append("\"")
-            .toString();
+        String.format(
+            "# This query represents dataset \"%s\" for domain \"%s\" and was generated for %s",
+            dataSetName, domainAsString, cdrVersionName);
```

Comment:
We should also include the workspace and notebook name in the comment, as this will get copied and pasted. The email of the researcher would also be helpful in order to establish provenance.

As written, they'll have to contact support to find the actual source of the query, and if we only include names, it's going to be even harder (since they don't uniquely identify datasets and domain names may change from one CDR version to the next (though rarely)).

Reply:
I'd discuss with product other likely entries here and double check that the sentence format still makes sense as you add things. Our users can all read JSON, and would be less likely to delete or edit it.